test/SlowTests/Issues/RavenDB-18554.cs

```diff
@@ -0,0 +1,212 @@
+                        {
+                            session.Query<Categoroies_Details.Entitiy, Categoroies_Details>()
+                                .Where(x => x.Id == categoryId)
+                                .ProjectInto<Categoroies_Details.Entitiy>()
+                                .ToList();
+                        }
+                    }
+                    catch (Exception e1)
+                    {
+                        e = e1;
+                    }
+                };
+
+                var operation = await store.Maintenance.Server.SendAsync(new CompactDatabaseOperation(settings));
+                await operation.WaitForCompletionAsync();
+
+                if (!cluster)
+                {
+                    Assert.NotNull(e);
+                    Assert.True(e is DatabaseDisabledException);
+                }
+                else
+                {
+                    Assert.Null(e); // Faliover
+                }
+            }
+        }
+
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public async Task QueriesShouldFailoverIfIndexIsCompacting(bool cluster = false)
+        {
+            Options storeOptions;
+            RavenServer leader = null;
+            if (cluster)
+            {
+                List<RavenServer> nodes;
+                (nodes, leader) = await CreateRaftCluster(2);
+                storeOptions = new Options { Server = leader, ReplicationFactor = nodes.Count, RunInMemory = false };
+            }
+            else
+            {
+                storeOptions = new Options { RunInMemory = false };
+            }
+
+            using (var store = GetDocumentStore(storeOptions))
+            {
+                // Prepare Server For Test
+                string categoryId;
+                using (var session = store.OpenAsyncSession())
+                {
+                    Category c = new Category { Name = $"n0", Description = $"d0" };
+                    await session.StoreAsync(c);
+                    await session.SaveChangesAsync();
+                    categoryId = c.Id;
+                }
+
+                var index = new Categoroies_Details();
+                index.Execute(store);
+                Indexes.WaitForIndexing(store);
+
+                string[] indexNames = { index.IndexName };
+                CompactSettings settings = new CompactSettings { DatabaseName = store.Database, Documents = true, Indexes = indexNames };
+                var database = cluster ? await GetDatabase(leader, store.Database) : await GetDatabase(store.Database);
+
+                // Test
+                Exception e = null;
+                database.ForTestingPurposesOnly().IndexCompaction = () =>
+                {
+                    try
+                    {
+                        using (var session = store.OpenSession())
+                        {
+                            session.Query<Categoroies_Details.Entitiy, Categoroies_Details>()
```

Comment:
Check that we got the correct result here (for cluster)

Reply:
Done